### ISSUE TITLE
Why is the Updater calling for google-analytics?

### ISSUE THREAD

**Issue by JPVenson** [Rem0o/FanControl.Releases#1313]
There is no Privacy statement what the downloader submits and how it is used. Also blocking this crashes the updater.

![image](https://user-images.githubusercontent.com/6794763/203414999-16f340ad-bb49-43a7-b3e4-5a256b79d92e.png)

**Comment by Rem0o** [comment]
See https://github.com/Rem0o/FanControl.Releases/issues/1296

GA is used as an update counter per version. There is no user data being shared, no user id, no tracking anything.  

That's the json being sent.  

```json
{
    "client_id" : "Updater",
    "events" : [
        {
            "name" : "update",
            "params":
            {
                 "version" : 137,
            }
         }
       ]
    }
}
```

Before I used a bitly link as a redirect to the file zip url, exact same result, but I phased that out. 

As for the crash itself, I thought I fixed it yesterday. Basically it should any failure and move along. 
When was the updater last modified on your machine?

**Comment by JPVenson** [comment]
Well yes i assume, any data collection to those services must be explicitly stated in the EULA. Thats the reason why in all setups you have one :-). That is the problem with FanControl, we give the program a big benefit of a doubt because it is not open source, so i have to spin up fiddler to verify that if i not just want to disable it in pihole. But also the fact that it is not advertised and does crash is not on the "trustworthy" side. Please do not think i assume malice here or something like that it is just "strange".

So in that case OpenSourcing the Updater would be anyways a step in the right direction, but IMOP creating an EULA if you still want to use GA and in addition making it optional would be the right choise.

"Updater.exe" last modifed 22.11.2022 21:29 
Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          F8A997B119428BD083A5C5497C1F887C07FAB436167520B8D5A6C15BA8480CB9       D:\FanControl\Updater.exe

**Comment by JPVenson** [comment]
https://marketingplatform.google.com/about/analytics/terms/gb/

It is also stated in the GA TOS that you have to provide an Privacy policy the user must accept:
7. Privacy

> You will have and abide by an appropriate Privacy Policy and will comply with all applicable laws, policies and regulations relating to the collection of information from Users. You must post a Privacy Policy and that Privacy Policy must provide notice of Your use of cookies, identifiers for mobile devices (e.g., Android Advertising Identifier or Advertising Identifier for iOS) or similar technology used to collect data. You must disclose the use of Google Analytics, and how it collects and processes data.

**Comment by Rem0o** [comment]
^ In any case, would be the occasion to also copy the license stuff from all the various libraries into FanControl directly, right now only having links to the repos, which again is iffy at best, call me lazy.  I will queue that up on the top of my todos.

**Comment by Rem0o** [comment]
@JPVenson 

Made a really basic EULA / privacy policy.   It will be presented on first launch of the app, then available from the side menu.  I think I covered the basics that I wanted to cover. I don't want to go overboard with lawyer speak, so I kept it in layman terms. 
Would you mind making a quick review?

**Comment by JPVenson** [comment]
@Rem0o I had a look into the draft and for the intend it is sufficient in my opinion (#NotLegalAdvice i am not an Lawer). Generally, i always link the [GA license](https://policies.google.com/privacy?hl=en) and (not applicable here) the AppInsights [privacy policy](https://learn.microsoft.com/en-us/azure/azure-monitor/app/data-retention-privacy) as this is sufficient enough for my Jurisdiktion and provides much better explanation what happens with that data. 

I would really not say "and can't be personified to a specific user" because that is technically not correct, as you do not quite have control over GA data. I would formulate it as "will not be used to identify any specific user" or so.

Otherwise it seems fine.

**Comment by Rem0o** [comment]
> they will use it to cross-device track you

I specifically added the flag "anonymize IP" and disabled all tracking toggle that I could. But I guess you are right, Google gonna be Google in the end.  Will link those 2, polish it up a bit, and push it into the next update.   Thanks!

**Comment by Rem0o** [comment]
See V138

**Comment by JPVenson** [comment]
Updater Confirmand working as of today. However after the update i did not see a popup with the EULA.

**Comment by Rem0o** [comment]
^ The popup is linked to "First Launch", so if you download fresh, or kill the cache file, you'll get it.
